docs/developer/architecture/user_customisations.rst

```diff
@@ -0,0 +1,60 @@
+User Customisations and Persistence
+===================================
+
+Context
+-------
+
+MetalK8s comes with a set of tools and services that may need to be configured
+on site. At the same time, we don't want the administrator of the cluster to
+master each and every service of the cluster. We also don't want to allow all
+kind of customisations since it would make the system even more complex to test
+and maintain over time.
+
+In addition to those services, MetalK8s deployment may have to be adapted
+depending on the architecture of the platform or depending on the different
+use cases and applications running on the top of it.
+
+It can be:
+
+- The BootstrapConfig,
+- The various roles and taints we set on the node objects of the cluster
+- The configurations associated to solutions, such as the list of available
+  solutions, the environments and namespaces created for a solution
+
+Be it services or MetalK8s customisations, we need to ensure it is persisted
+and resilient to various type of events such as node reboot, upgrade, downgrade
+, backup, restore.
+
+.. _usercusto-requirements:
+
+Requirements
+------------
+
+As a MetalK8s administrator, I have access to a finite list of settings I can
+customise on-site in order to match with my environment specificities:
+
+- list of static users and credentials configured in Dex
+- Integration with an external IDP configuration in Dex
+- Alert notification configuration in Alert Manager
+- New Grafana dashboards or new Grafana datasources
```

Comment:
@NicolasT @gdemonet Should the persistence of Grafana dashboard be handled like this? From the Prometheus operator charts, it looks like the production recommended way was to use volumes and PVC's just like we did for Alertmanager & Prometheus.

Reply:
The requirement exists however. It may imply that the design also includes "exceptions" (e.g. provide a PV for Grafana, although I'm not sure how that fits with dashboards configured in ConfigMaps...), or simply acknowledges that this requirement is not handled (which could obviously be dismissed as not being acceptable).